source/game_sa/PathFind.cpp

```diff
@@ -584,10 +586,38 @@ void CPathFind::MarkRoadNodeAsDontWander(float x, float y, float z) {
 }
 
 // 0x452820
-void CPathFind::SwitchRoadsOffInAreaForOneRegion(float xMin, float xMax, float yMin, float yMax, float zMin, float zMax, bool bLowTraffic, uint8 nodeType, int32 areaId,
-                                                 uint8 bUnused) {
-    return plugin::CallMethod<0x452820, CPathFind*, float, float, float, float, float, float, bool, char, int32, bool>(this, xMin, xMax, yMin, yMax, zMin, zMax, bLowTraffic,
-                                                                                                                       nodeType, areaId, bUnused);
+void CPathFind::SwitchRoadsOffInAreaForOneRegion(float xMin, float xMax, float yMin, float yMax, float zMin, float zMax, bool bSwitchOff, bool bCars, int areaId, bool bBackToOriginal) {
+    assert(areaId >= 0 && areaId < NUM_PATH_MAP_AREAS);
+
+    if (!IsAreaLoaded(areaId)) {
+        return;
+    }
+
+    auto start = bCars ? 0 : m_anNumVehicleNodes[areaId];
+    auto end   = bCars ? m_anNumVehicleNodes[areaId] : m_anNumNodes[areaId];
+
+    for (auto nodeIdx = start; nodeIdx < end; ++nodeIdx) {
+        CPathNode& node     = m_pPathNodes[areaId][nodeIdx];
+        const auto position = node.GetPosition();
+
+        if (position.x < xMin || position.x > xMax || position.y < yMin || position.y > yMax || position.z < zMin || position.z > zMax) {
+            continue;
+        }
+        if (!ThisNodeHasToBeSwitchedOff(&node) || node.m_isSwitchedOff == (bBackToOriginal ? node.m_isSwitchedOffOriginal : bSwitchOff)) {
+            continue;
+        }
+        CPathNode* next1{};
+        CPathNode* next2{};
+
+        SwitchOffNodeAndNeighbours(&node, next1, &next2, bSwitchOff, bBackToOriginal);
+
+        while (const auto nextNode = next1) {
+            SwitchOffNodeAndNeighbours(nextNode, next1, nullptr, bSwitchOff, bBackToOriginal);
+        }
+        while (const auto nextNode = next2) {
+            SwitchOffNodeAndNeighbours(nextNode, next2, nullptr, bSwitchOff, bBackToOriginal);
+        }
```

Comment:
Perhaps this is more understandable, because it took me a while to figure out what's exactly happening.
```suggestion
        for (auto iter = next1; iter;) {
            SwitchOffNodeAndNeighbours(iter, iter, nullptr, bSwitchOff, bBackToOriginal);
        }
        for (auto iter = next2; iter;) {
            SwitchOffNodeAndNeighbours(iter, iter, nullptr, bSwitchOff, bBackToOriginal);
        }
```